src/action.js

```diff
@@ -44,25 +46,31 @@ module.exports = async function run() {
       return logWarning('Not a dependabot PR, skipping.')
     }
 
+    const prDiff = await client.getPullRequestDiff(pr.number)
+    const moduleChanges = getModuleVersionChanges(prDiff)
+
     if (TARGET !== targetOptions.any) {
-      logInfo(`Checking if PR title [${pr.title}] has target ${TARGET}`)
-      const isTargetMatchToPR = checkTargetMatchToPR(pr.title, TARGET)
+      logInfo(`Checking if the changes in the PR can be merged`)
 
+      const isTargetMatchToPR = checkModuleVersionChanges(moduleChanges, TARGET)
       if (!isTargetMatchToPR) {
         return logWarning('Target specified does not match to PR, skipping.')
       }
     }
 
-    const { name: pkgName, version } = getPackageDetails(pr)
-    const upgradeMessage = `Cannot automerge github-action-merge-dependabot ${version} major release.
-  Read how to upgrade it manually:
-  https://github.com/fastify/github-action-merge-dependabot#how-to-upgrade-from-2x-to-new-3x`
-
-    if (EXCLUDE_PKGS.includes(pkgName)) {
-      return logInfo(`${pkgName} is excluded, skipping.`)
+    const changedExcludedPackages = EXCLUDE_PKGS.filter((pkg) => pkg in moduleChanges)
+    if (changedExcludedPackages.length > 0) {
+      return logInfo(`${changedExcludedPackages.length} package(s) excluded: \
+${changedExcludedPackages.join(', ')}. Skipping.`)
     }
 
-    if (pkgName === 'github-action-merge-dependabot' && isMajorRelease(pr)) {
+    const thisModuleChanges = moduleChanges['github-action-merge-dependabot']
```

Comment:
let's please extract this hardcoded string into a constant or, even better, grab it from some place where it already exists, e.g. pkg json?

Reply:
Done, @simoneb. Thanks for the review